File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/cpp_namespace_runme.m
# Note: This example assumes that namespaces are flattened
cpp_namespace

n = cpp_namespace.fact(4);
if (n != 24)
    error("Bad return value!")
endif

if (cpp_namespace.cvar.Foo != 42)
    error("Bad variable value!")
endif

t = cpp_namespace.Test();
if (!strcmp(t.method(),"Test::method"))
    error("Bad method return value!")
endif

if (!strcmp(cpp_namespace.do_method(t),"Test::method"))
    error("Bad return value!")
endif

if (!strcmp(cpp_namespace.do_method2(t),"Test::method"))
    error("Bad return value!")
endif
    
cpp_namespace.weird("hello", 4);

clear t;

t2 = cpp_namespace.Test2();
t3 = cpp_namespace.Test3();
t4 = cpp_namespace.Test4();
t5 = cpp_namespace.Test5();

if (cpp_namespace.foo3(42) != 42)
    error("Bad return value!")
endif

if (!strcmp(cpp_namespace.do_method3(t2,40),"Test2::method"))
    error("Bad return value!")
endif

if (!strcmp(cpp_namespace.do_method3(t3,40),"Test3::method"))
    error("Bad return value!")
endif

if (!strcmp(cpp_namespace.do_method3(t4,40),"Test4::method"))
    error("Bad return value!")
endif

if (!strcmp(cpp_namespace.do_method3(t5,40),"Test5::method"))
    error("Bad return value!")
endif

    

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/cpp_namespace_runme.lua
require("import")	-- the import fn
import("cpp_namespace")	-- import lib into global
cn=cpp_namespace --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

assert(cn.fact(4) == 24)
assert(cn.Foo == 42)

t1 = cn.Test()
assert(t1:method() == "Test::method")

cn.weird("t1", 4)

assert(cn.do_method(t1) == "Test::method")
assert(cn.do_method2(t1) == "Test::method")

t2 = cn.Test2()
assert(t2:method() == "Test2::method")


assert(cn.foo3(5) == 5)

assert(cn.do_method3(t2, 7) == "Test2::method")
